skbio/io/fasta.py

```diff
@@ -557,6 +563,10 @@
 
 @register_sniffer('fasta')
 def _fasta_sniffer(fh):
+
+    if _count_blank_lines(fh) > 5:
```

Comment:
The current implementation of `_count_blank_lines` will read an entire whitespace-only file since it only stops reading if it finds a non-whitespace line or reaches the end of the file. Can you modify the way the sniffer works so that it'll bail as soon as it reaches 5 blank lines?